includes/helper/class-wp-job-manager-helper-api.php

```diff
@@ -77,13 +77,44 @@ public function bulk_update_check( array $plugins ) {
 	 * Sends and receives data related to plugin information from the WPJobManager.com API.
 	 *
 	 * @param array $args  The arguments to pass to the endpoint.
-	 * @return array|false The response, or false if the request failed.
+	 * @return \stdClass|false The response, or false if the request failed.
 	 */
 	public function plugin_information( $args ) {
-		$args            = wp_parse_args( $args );
-		$args['wc-api']  = 'wp_plugin_licencing_update_api';
-		$args['request'] = 'plugininformation';
-		return $this->request( $args );
+		$args = wp_parse_args( $args );
+		$data = $this->request_endpoint(
+			'wp-json/wpjmcom-licensing/v1/plugin-information',
+			[
+				'method' => 'GET',
+				'body'   => [
+					'site_url'     => $this->get_site_url(),
+					'license_key'  => $args['license_key'],
+					'product_slug' => $args['api_product_id'],
+				],
+			]
+		);
+		if ( ! is_array( $data ) ) {
+			return false;
+		}
```

Comment:
Here we have a difference in behavior compared to the current WPJM version: Currently, when the legacy endpoint on WPJMCOM receives a `product_slug` that doesn't exist, it returns an empty `stdClass` (but still with all the correct format). In every other case, we don't return an error per se.

Here, that doesn't happen: When the `product_slug` doesn't match anything we expect, we return only `false`, which is handled by WP as if the request failed (and hence we should return what WP already knows about it: nothing). 

Technically, WPJM already has protections in place to guarantee that this method won't even be called in case we're not handling a WPJM addon (by looking at the `WPJM-Product` on the plugin header), so I don't think this is a big deal. However, it is worth to keep in mind, anyway